indra/assemblers/indranet/assembler.py

```diff
@@ -281,9 +369,173 @@ def make_df(self, exclude_stmts=None, complex_members=3,
         df = df.where((pd.notnull(df)), None)
         return df
 
+    def make_model_by_preassembly(self, exclude_stmts=None, complex_members=3,
+                                  graph_type='multi_graph', sign_dict=None,
+                                  belief_scorer=None, weight_flattening=None,
+                                  extra_columns=None):
+        """Assemble an IndraNet graph object.
+
+        Parameters
+        ----------
+        exclude_stmts : list[str]
+            A list of statement type names to not include in the graph.
+        complex_members : int
+            Maximum allowed size of a complex to be included in the graph.
+            All complexes larger than complex_members will be rejected. For
+            accepted complexes, all permutations of their members will be added
+            as edges. Default is `3`.
+        graph_type : str
+            Specify the type of graph to assemble. Chose from 'multi_graph'
+            (default), 'digraph', 'signed'. Default is `multi_graph`.
+        sign_dict : dict
+            A dictionary mapping a Statement type to a sign to be used for
+            the edge. This parameter is only used with the 'signed' option.
+            See IndraNet.to_signed_graph for more info.
+        belief_scorer : Optional[indra.belief.BeliefScorer]
+            Instance of BeliefScorer class to use in calculating edge
+            probabilities. If None is provided (default), then the default
+            scorer is used.
+        weight_flattening : function(networkx.DiGraph)
+            A function taking at least the graph G as an argument and
+            returning G after adding edge weights as an edge attribute to the
+            flattened edges using the reserved keyword 'weight'.
+
+            Example:
+
+            >>> def weight_flattening(G):
+            ...     # Sets the flattened weight to the average of the
+            ...     # inverse source count
+            ...     for edge in G.edges:
+            ...         w = [1/s['evidence_count']
+            ...             for s in G.edges[edge]['statements']]
+            ...         G.edges[edge]['weight'] = sum(w)/len(w)
+            ...     return G
+
+        Returns
+        -------
+        model : IndraNet
+            IndraNet graph object.
+        """
+        # Filter out statements with one agent or with None subject
+        stmts = [stmt for stmt in self.statements if len(
+            [ag for ag in stmt.agent_list() if ag is not None]) > 1]
```

Comment:
You can use `stmt.real_agent_list()` instead of `stmt.agent_list() if ag is not None`